cinema/views.py

```diff
@@ -124,15 +161,23 @@
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound()
```

Comment:
In `OrderViewSet`, the `get_permissions` method raises a `NotFound` exception for unsupported actions. Consider using `MethodNotAllowed` to indicate that the method is not supported.